libraries/apollo-api/src/commonMain/kotlin/com/apollographql/apollo3/api/unsafeCastOrCast.kt

```diff
@@ -0,0 +1,14 @@
+package com.apollographql.apollo3.api
+
+expect inline fun <reified T>  Any.unsafeCastOrCast(): T?
+
+/**
+ * Uses `unsafeCast` on JS or a regular `as?` cast on other platforms. Note that
+ * this means that JS platforms will crash at runtime if the cast fails, whereas
+ * other (statically) typed targets will return a null value.
+ *
+ * @return
+ */
+inline fun <reified T> Any.defaultUnsafeCastOrCast(): T? {
+  return this as? T
```

Comment:
I think it makes more sense to default to `as` here? Because:

1. Kotlin refers to `as` as "unsafe" cast in its [doc](https://kotlinlang.org/docs/typecasts.html#unsafe-cast-operator)
2. More generally, if there's a disconnect between `__typename` and the shape of the response, it's an error and it's best to catch it as soon as possible?

Reply:
All in all, I think I would go with something like so: 

```
@ApolloExperimental
inline fun <reified T> Any.apolloUnsafeCast(): T 
```

Or do you explicitely need the `as?` behaviour on non-JS?